### PR TITLE
VDI-1714 QA backup activity added a passphase switch for UCP backup

### PR DESCRIPTION
**Pull request by moyni77**
Backup of UCP now requires a passphrase to be passed to encrypt data and comply with security best practices.

### CODE REVIEW

**Comment by gabrielmcg** [comment]
Can the password be stored in the vault, instead of in the clear?

**Comment by moyni77** [comment]
Making alteration suggested by Gabriel.

**Comment by moyni77** [comment]
Ready once again for review.

**Comment by moyni77** [comment]
I've made the alteration in restore_ucp.sh.j2 template to include passphrase, I'm just in the middle of testing restore.yml.

File: group_vars/backups [hunk 1]
```diff
@@ -15,14 +15,14 @@
 #
 backup_swarm: "/root/restore/backup_swarm.tgz"
 backup_ucp: "/root/restore/backup_ucp.tgz"
-backup_dtr_meta: "/root/restore/backup_dtr_meta.tgz"
-backup_dtr_data: "/root/restore/backup_dtr_data.tgz"
+backup_dtr_meta: "/root/restore/backup_dtr_meta.vars.tgz"
```
Comment (chris7444) [inline group_vars/backups]: You tested with backup_dtr_meta poiting to the .vars.tgz file ?

File: group_vars/backups [hunk 2]
```diff
@@ -15,14 +15,14 @@
...
+backup_dtr_data: "/root/restore/backup_dtr_meta.tgz"
```
Comment (chris7444) [inline group_vars/backups]: same here. backup_dtr_data should point to something called backup_dtr_data.tgz. That is the file which contains the DTR images. It seems you are pointing to a file that contains the DTR metatadata.

**Comment by moyni77** [comment]
Note: backups file has been corrected and the restore is being re-run. Will provide feedback once complete. I have had to re-create my llb hosts and reflect what I have in my own ucp client bundle as new load ballancer hosts weren't part of the subject alternative name list therefore not being picked up.

**Comment by moyni77** [comment]
Restore was successful UCP,DTR and workers recovered. Please approve.

File: group_vars/backups (hunk 2 above)
Comment (moyni77) [reply group_vars/backups]: backup_ucp: "/root/restore/backup_ucp_w13erp1dwod5cqu3bmptch9b2_nm-ucp01_2019_02_12_085147.tgz"
backup_dtr_meta: "/root/restore/backup_dtr_meta_002cf30a42f5_nm-dtr01_2019_02_12_085337.tgz"
backup_dtr_data: "/root/restore/backup_dtr_data_002cf30a42f5_nm-dtr01_2019_02_12_085432.tgz"

**Comment by chris7444** [comment]
@moyni77 On my side, images are not restored. The fact that the playbooks finishes successfully does not mean that the restore was successful. I cannot approve unless you tell me you have verified that users are restored, and DTR images are restored

**Comment by moyni77** [comment]
Users, and the repositories recover fine. However the images-folder isn't populated, This isn't in connection with the alteration I've had to make to include a passphrase for UCP. Investigation as to why images aren't restore is in progress.

**Comment by moyni77** [comment]
Using DTR version 2.6.1 images are restored successfully and tags 3.4. and 2.6 are visible in our previously created images repository. Please can you now approve this pull request. For the interim release we will pin to 2.6.1.

**Comment by chris7444** [comment]
let's see what the other say tomorrow

**Comment by gabrielmcg** [comment]
I am happy to commit this as the passphrase is working correctly - we can open another JIRA/PR for the problems with restore

**Comment by moyni77** [comment]
Created https://jira.simplivt.local/browse/VDI-1806 for the DTR image restore defect.

**Comment by chris7444** [comment]
approving given the jira you created

**Pull request merged**